### DISCUSSION TITLE
Query parameters inside classes does not generate examples

### DISCUSSION THREAD

**Discussion by Behoston** [fastapi/fastapi#9953] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Query, Depends
from pydantic import BaseModel

app = FastAPI(docs_url='/')


class CommonSearchParams(BaseModel):
    limit: int = Query(ge=1, le=50, example=15)
    offset: int = Query(ge=0, examples=[0])
    order_by: str = Query(default="created_date")


@app.get("/search")
async def search(
        common_search_params: CommonSearchParams = Depends(CommonSearchParams),
        name: str = Query(example='Andrzej'),
        last_name: str = Query(examples=['Kowalski', 'Nowak']),
):
    return {"message": "Hello World"}
```

### Description

When Query parameters are grouped in BaseModel class, it only produces default values and do not contain examples in schema. There is no difference between example and examples - both does not work.

BTW swagger UI does not show any example values when `examples` used. 

### Operating System

Linux

### FastAPI Version

0.100.0

### Python Version

3.8.10

**Comment by accountForIssues** [comment]
I'm pretty sure this is due to the changes introduced in OpenAPI spec version 3.1 and the readers not fully supporting it.

I use a dataclass decorator for the request params and for me, the schema json actually includes examples just fine.

But, Swagger doesn't show anything and Redoc does show them. You should give the latest FastAPI with Redoc a try.

I think at this point, FastAPI should support examples of both 3.0 and 3.1 specs rather than waiting for Swagger, Redoc, etc. to catch up.

**Comment by tiangolo** [comment, marked as answer]
Grouping parameters in Pydantic models is actually not supported (yet). 😅 

It's a hack that worked by chance, but it hasn't been supported, although I plan to support doing something similar in the future. 🤓
